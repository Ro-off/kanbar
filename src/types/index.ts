import { SVGProps } from "react";

import { ColumnIds } from "@/components/taskRow";

export type IconSvgProps = SVGProps<SVGSVGElement> & {
  size?: number;
};

export interface Task {
  id: string;
  description?: string;
  newField?: boolean;
  columnId: ColumnIds;

  [key: string]: any;
}

export interface AddTaskAction {
  type: "ADD_TASK";
  payload: ColumnIds;
}

export interface RemoveTaskAction {
  type: "REMOVE_TASK";
  payload: string;
}

export interface UpdateTaskAction {
  type: "UPDATE_TASK";
  payload: {
    id: string;
    data: Partial<Task>;
    index?: number;
  };
}

export type TaskAction = AddTaskAction | RemoveTaskAction | UpdateTaskAction;

export type TasksState = {
  tasks: Task[];
};

export interface TaskCardProps extends Task {
  isEditing?: boolean;
  index?: number;
}
