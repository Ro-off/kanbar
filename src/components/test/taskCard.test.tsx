import { expect, it } from "vitest";
import { render } from "@testing-library/react";
import { TaskCard } from "@components";

import { TaskCardProps } from "@/types";

const taskCardTestProps: TaskCardProps = {
  id: "1234-1234-1234-1234",
  columnId: "todo",
  description: "This is a test task card",
  newField: false,
  index: 0,
};

it("Check taskCard snapshot", () => {
  const { asFragment } = render(<TaskCard {...taskCardTestProps} />);

  expect(asFragment()).toMatchSnapshot();
});
