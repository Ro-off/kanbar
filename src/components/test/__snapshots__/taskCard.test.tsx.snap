// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check TaskCard snapshot 1`] = `
<DocumentFragment>
  <div
    aria-describedby=""
    aria-disabled="false"
    aria-roledescription="sortable"
    class=" relative z-50"
    role="button"
    tabindex="0"
  >
    <div
      class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none mx-4 my-2"
      tabindex="-1"
    >
      <div
        class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full opacity-disabled pointer-events-none is-filled"
          data-disabled="true"
          data-filled="true"
          data-filled-within="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-8 min-h-8 px-2 rounded-small pointer-events-none !h-auto transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background py-2 bg-transparent shadow-none is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <div
              class="inline-flex w-full h-full box-border items-start group-data-[has-label=true]:items-start"
              data-slot="inner-wrapper"
            >
              <textarea
                aria-label=" "
                class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small resize-none data-[hide-scroll=true]:scrollbar-hide group-data-[has-value=true]:text-default-foreground transition-height !duration-100 motion-reduce:transition-none is-filled pe-0"
                data-filled="true"
                data-filled-within="true"
                data-hide-scroll="true"
                data-slot="input"
                disabled=""
                id="react-aria-:r0:"
              >
                This is a test task card
              </textarea>
            </div>
          </div>
        </div>
      </div>
      <div
        class="p-3 h-auto w-full items-center overflow-hidden color-inherit subpixel-antialiased rounded-b-large flex justify-between"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default/40 text-default-700 data-[hover=true]:opacity-hover btn btn-secondary"
          data-react-aria-pressable="true"
          tabindex="0"
          type="button"
        >
          Delete
        </button>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default/40 text-default-700 data-[hover=true]:opacity-hover btn btn-primary"
          data-react-aria-pressable="true"
          tabindex="0"
          type="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Check taskCard snapshot 1`] = `
<DocumentFragment>
  <div
    aria-describedby=""
    aria-disabled="false"
    aria-roledescription="sortable"
    class=" relative z-50"
    role="button"
    tabindex="0"
  >
    <div
      class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none mx-4 my-2"
      tabindex="-1"
    >
      <div
        class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full opacity-disabled pointer-events-none is-filled"
          data-disabled="true"
          data-filled="true"
          data-filled-within="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-8 min-h-8 px-2 rounded-small pointer-events-none !h-auto transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background py-2 bg-transparent shadow-none is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <div
              class="inline-flex w-full h-full box-border items-start group-data-[has-label=true]:items-start"
              data-slot="inner-wrapper"
            >
              <textarea
                aria-label=" "
                class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 data-[type=color]:rounded-none file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small resize-none data-[hide-scroll=true]:scrollbar-hide group-data-[has-value=true]:text-default-foreground transition-height !duration-100 motion-reduce:transition-none is-filled pe-0"
                data-filled="true"
                data-filled-within="true"
                data-hide-scroll="true"
                data-slot="input"
                disabled=""
                id="react-aria-:r0:"
              >
                This is a test task card
              </textarea>
            </div>
          </div>
        </div>
      </div>
      <div
        class="p-3 h-auto w-full items-center overflow-hidden color-inherit subpixel-antialiased rounded-b-large flex justify-between"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default/40 text-default-700 data-[hover=true]:opacity-hover btn btn-secondary"
          data-react-aria-pressable="true"
          tabindex="0"
          type="button"
        >
          Delete
        </button>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default/40 text-default-700 data-[hover=true]:opacity-hover btn btn-primary"
          data-react-aria-pressable="true"
          tabindex="0"
          type="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
